index_trade.html

```diff
@@ -0,0 +1,406 @@
+    });      
+      
+  });
+
+    
+  function Chart(xvar, xtitle, yvar, ytitle) {
+      var chart = this;
+      
+//      chart.variable = variable;
+      chart.xvar = xvar;
+      chart.yvar = yvar;
+
+    // SVG
+      chart.svg = d3.select('#chart')
+      .append('svg')
+      .attr('width', width + margin.right + margin.left)
+      .attr('height', height + margin.top + margin.bottom)
+      .append('g')
+      .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+      
+    // SCALES
+//      var xmin = d3.min(data, function (d) { return d.trade; }) - 500;
+      var xmin = d3.min(data, function (d) { return d[xvar]; }) - 500;      
+      
+//      x = d3.scale.linear()
+      
+      chart.x = d3.scale.linear()
+        .domain([xmin, -xmin])
+        .range([0, width])
+        .nice();
+      
+//    y = d3.scale.linear()      
+      chart.y = d3.scale.linear()
+      .domain([0, d3.max(data, function (d) { return d[yvar]; })])    
+      .range([height, 0])    
+      .nice();
+      
+    // AXES
+      var xAxis = d3.svg.axis()
+      .scale(chart.x)
+      .tickSize(-height);
+      
+      var yAxis = d3.svg.axis()
+      .scale(chart.y)
+      .tickSize(-width)
+      .orient('left');   
+      
+//      svg.append('g')
+      chart.svg.append('g')      
+          .attr('class', 'x axis')
+          .call(xAxis)
+          .attr('transform', 'translate(0,' + (height) + ')')
+        .append('text')
+          .attr('x', width/2)
+          .attr('y', 70)
+          .style('text-anchor', 'middle')
+          .text(xtitle);            
+      
+//      svg.append('g')
+      chart.svg.append('g')
+          .attr('class', 'y axis')
+          .call(yAxis)
+        .append('text')
+          .attr('transform', 'rotate(-90)')
+          .attr('y', -60)
+          .attr('x', -height/2)      
+          .style('text-anchor', 'middle')      
+          .text(ytitle);           
+      
+      // GRID LINES
+      
+/*      var ygrid = svg.append('g')
+      .attr('class', 'y axis')
+      .call(yAxis);
+      
+      ygrid.selectAll('g').filter(function(d) { return d; })
+        .classed('minor', true);*/
+
+      // HISTOGRAM
+      
+      // x hist
+      
+//    var xmap = data.map(function(d) { return d.trade; })
+//    var xhist = d3.layout.histogram().bins(x.ticks())(xmap);   
+      
+
+      
+//      console.log('hist', xhist);
+      
+
+//      console.log('bar', barWidth, x.ticks());
+      
+     
+      
+      // y hist
+      
+//    var ymap = data.map(function(d) { return d.rate; })
+//    var yhist = d3.layout.histogram().bins(y.ticks())(ymap);      
+      
+//      console.log('rate', yhist.reverse());
+
+      chart.xtool = d3.tip()
+      .attr('class', 'tooltip')
+      .offset([-6, 0])
+      .html(function(d) { return d.length ; });
+
+      chart.ytool = d3.tip()
+      .attr('class', 'tooltip')
+      .offset([10, 40])
+      .html(function(d) { return d.length ; });
+
```

Comment:
So there's some nifty functionality in the d3.tip function to place the tooltip to the left/right/bottom of an element instead of just above it (the default). So instead of the offset, if you use:
```js
      chart.ytool = d3.tip()
      .attr('class', 'tooltip')
      .offset([0, 0])
      .direction('e')
      .html(function(d) { return d.length ; });
```


More on that: https://github.com/Caged/d3-tip/blob/master/docs/positioning-tooltips.md